baybe/surrogates/validation.py

```diff
@@ -53,52 +57,63 @@ def validate_custom_architecture_cls(model_cls: type) -> None:
         )
 
 
-def get_model_params_validator(model_init: Callable | None = None) -> Callable:
-    """Construct a validator based on the model class.
+# Create a strict type validation converter
+type_validation_converter = converter = cattrs.GenConverter(forbid_extra_keys=True)
+"""Converter used for strict type validation."""
+
+
+configure_union_passthrough(int | float | str | None, type_validation_converter)
+
+
+def _strict_int_structure_hook(obj: Any, _: type[int]) -> int:
```

Comment:
Just FYI: since you nicely typed the functions, you can actually use the slightly more elegant modern registration syntax using decorators (likewise for the others)
```suggestion
@type_validation_converter.register_structure_hook
def _strict_int_structure_hook(obj: Any, _: type[int]) -> int:
```

Reply:
changed